logbook-logstash-logback-encoder/src/test/java/org/zalando/logbook/lle/LogbackLogstashTest.java

```diff
@@ -0,0 +1,143 @@
+        
+        final String correlationId = "3ce91230-677b-11e5-87b7-10ddb1ee7671";
+        final HttpRequest request = MockHttpRequest.create()
+                .withProtocolVersion("HTTP/1.0")
+                .withOrigin(REMOTE)
+                .withPath("/test")
+                .withQuery("limit=1")
+                .withHeaders(MockHeaders.of(
+                        "Accept", "application/json",
+                        "Date", "Tue, 15 Nov 1994 08:12:31 GMT"))
+                .withContentType("application/json")
+                .withBodyAsString("{\"person\":{\"name\":\"Thomas\"}}");
+
+        sink.write(new SimplePrecorrelation(correlationId, systemUTC()), request);
+
+        for(String last : new String[] {StaticAppender.getLastStatement(), PrettyPrintingStaticAppender.getLastStatement()} ) {
+            with(last)
+                .assertThat("$.http.origin", is("remote"))
+                .assertThat("$.http.type", is("request"))
+                .assertThat("$.http.correlation", is(correlationId))
+                .assertThat("$.http.protocol", is("HTTP/1.0"))
+                .assertThat("$.http.remote", is("127.0.0.1"))
+                .assertThat("$.http.method", is("GET"))
+                .assertThat("$.http.uri", is("http://localhost/test?limit=1"))
+                .assertThat("$.http.headers.*", hasSize(2))
+                .assertThat("$.http.headers['Accept']", is(singletonList("application/json")))
+                .assertThat("$.http.headers['Date']", is(singletonList("Tue, 15 Nov 1994 08:12:31 GMT")))
+                .assertThat("$.http.body.person.name", is("Thomas"));
+        }
+        
+        final HttpResponse response = MockHttpResponse.create()
+                .withProtocolVersion("HTTP/1.0")
+                .withOrigin(REMOTE)
+                .withHeaders(MockHeaders.of("Date", "Tue, 15 Nov 1994 08:12:31 GMT"))
+                .withContentType("application/json")
+                .withBodyAsString("{\"person\":{\"name\":\"Magnus\"}}");
+
+        sink.write(new SimpleCorrelation(correlationId, ofMillis(125)), request, response);
+
+        for(String last : new String[] {StaticAppender.getLastStatement(), PrettyPrintingStaticAppender.getLastStatement()} ) {
+            with(last)
+                .assertThat("$.http.origin", is("remote"))
+                .assertThat("$.http.type", is("response"))
+                .assertThat("$.http.correlation", is(correlationId))
+                .assertThat("$.http.protocol", is("HTTP/1.0"))
+                .assertThat("$.http.status", is(200))
+                .assertThat("$.http.headers.*", hasSize(1))
+                .assertThat("$.http.headers['Date']", is(singletonList("Tue, 15 Nov 1994 08:12:31 GMT")))
+                .assertThat("$.http.body.person.name", is("Magnus"))
+                .assertThat("$.http.duration", is(125));
+        }        
+    }
+
+    static class SimplePrecorrelation implements Precorrelation {
```

Comment:
You could just mock those.

Reply:
Mocked (originally copied from JsonHttpLogFormatterTest)

Reply:
Maybe worth doing it there too?